spec/tapioca/compilers/dsl/active_record_scope_spec.rb

```diff
@@ -0,0 +1,104 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe("Tapioca::Compilers::Dsl::ActiveRecordScope") do
+  before(:each) do
+    require "tapioca/compilers/dsl/active_record_scope"
+  end
+
+  subject do
+    Tapioca::Compilers::Dsl::ActiveRecordScope.new
+  end
+
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecord classes") do
+      assert_empty(subject.processable_constants)
+    end
+
+    it("gathers only ActiveRecord constants with no abstract classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+        end
+
+        class Product < ActiveRecord::Base
+          self.abstract_class = true
+        end
+
+        class User
+        end
+      RUBY
+
+      assert_equal(["Post"], constants_from(content))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generates RBI file for ActiveRecord classes with a scope field") do
```

Comment:
Can we also add a case where we show that we generate an empty RBI file if the model has no defined scopes?

Reply:
There will be no RBI file. In `our RBI generator` we are making sure not to generate any RBI file if there is no scope defined  with this line of code `return if scopes.blank?`

Reply:
That means the type won't be generated but there will be an empty RBI file as a result of running the generator.

It is the pipeline that prevents empty RBI files from not being written to disk, not the generator.

But either way, what we are trying to test is the case when the generator runs over a file that does not define any scopes.